twilight-model/src/channel/message/component/select_menu.rs

```diff
@@ -1,27 +1,59 @@
-use crate::channel::message::ReactionType;
+use crate::channel::{message::ReactionType, ChannelType};
 use serde::{Deserialize, Serialize};
 
-/// Dropdown-style [`Component`] that renders belew messages.
+/// Dropdown-style [`Component`] that renders below messages.
 ///
 /// [`Component`]: super::Component
 #[derive(Clone, Debug, Eq, Hash, PartialEq)]
 pub struct SelectMenu {
+    /// An optional list of channel types.
+    ///
+    /// This option is only used for [channel select menus](SelectMenuType::Channel).
+    pub channel_types: Option<Vec<ChannelType>>,
     /// Developer defined identifier.
     pub custom_id: String,
     /// Whether the select menu is disabled.
     ///
     /// Defaults to `false`.
     pub disabled: bool,
+    /// This select menu's type.
+    ///
+    /// [Text select menus](SelectMenuType::Text) *must* also set the `options` field.
+    pub kind: SelectMenuType,
     /// Maximum number of options that may be chosen.
     pub max_values: Option<u8>,
     /// Minimum number of options that must be chosen.
     pub min_values: Option<u8>,
-    /// List of available choices.
-    pub options: Vec<SelectMenuOption>,
+    /// A list of available options.
+    ///
+    /// This value is only used and required by [text select menus](SelectMenuType::Text).
```

Comment:
Earlier you use "option" and "field", here you use "value". Could we unify the language?

Reply:
I removed `value`, `field`, and `option` entirely from the documentation comments. Do you think this would address this concern?